src/index.rs

```diff
@@ -284,13 +312,17 @@ impl IndexTable {
 		key: u64,
 		address: Address,
 		source: &[u8],
+		// [Review] could be renamed to 'known_sub_index' or 'at_sub_index',
+		// in other sub_index calls, sub indexs is a starting query point, here it is insertion point.
 		sub_index: Option<usize>,
 		log: &mut LogWriter,
 	) -> Result<PlanOutcome> {
 		let chunk_index = key >> (64 - self.id.index_bits());
 		if address.as_u64() >= (1 << Entry::address_bits(self.id.index_bits())) {
 			// Address overflow
 			log::warn!(target: "parity-db", "{}: Address space overflow at {}: {}", self.id, chunk_index, address);
+			// [Review] Shouldn't it be unreachable (would mean all chunk full and one triggering reindex
+			// before)?
```

Comment:
I think I've experimented at some point with having `address_bits` = `index_bits`+ 6. So that if value size is  distributed unevenly, we may end up overflowing a single value table and run out of address space before running out of index space. This should not be possible with the current setup indeed, but I'd leave it there in case we decide to tweak these values again.